src/charm.py

```diff
@@ -1413,23 +1435,25 @@ def _on_update_status(self, _) -> None:
 
     def _was_restore_successful(self, container: Container, service: ServiceInfo) -> bool:
         """Checks if restore operation succeeded and S3 is properly configured."""
-        if service.current != ServiceStatus.ACTIVE:
-            if "restore-to-time" in self.app_peer_data and all(self.is_pitr_failed(container)):
-                logger.error(
-                    "Restore failed: database service failed to reach point-in-time-recovery target. "
-                    "You can launch another restore with different parameters"
-                )
-                self.log_pitr_last_transaction_time()
-                self.unit.status = BlockedStatus(CANNOT_RESTORE_PITR)
-                return False
+        if "restore-to-time" in self.app_peer_data and all(self.is_pitr_failed(container)):
+            logger.error(
+                "Restore failed: database service failed to reach point-in-time-recovery target. "
+                "You can launch another restore with different parameters"
+            )
+            self.log_pitr_last_transaction_time()
+            self.unit.status = BlockedStatus(CANNOT_RESTORE_PITR)
+            return False
 
-            if self.unit.status.message != CANNOT_RESTORE_PITR:
-                logger.error("Restore failed: database service failed to start")
-                self.unit.status = BlockedStatus("Failed to restore backup")
+        if (
+            service.current != ServiceStatus.ACTIVE
+            and self.unit.status.message != CANNOT_RESTORE_PITR
+        ):
```

Comment:
I suspect that this block is being reached earlier than the one above on Juju 2.9 because of the status message (`Failed to restore backup`) of the unit in the failed tests:
* https://github.com/canonical/postgresql-k8s-operator/actions/runs/10911425853/job/30284387686#step:31:699
* https://github.com/canonical/postgresql-k8s-operator/actions/runs/10911425853/job/30284388001#step:31:214
As I commented before on MM, the fix for that may be investigated later in another PR.

Reply:
That status is set [here after the PITR error check](https://github.com/canonical/postgresql-k8s-operator/blob/lucas/investigate-ci/src/charm.py#L1452), so I think the reason for this block running before the one above is because [this check failed when it shoud've succeeded](https://github.com/canonical/postgresql-k8s-operator/blob/lucas/investigate-ci/src/charm.py#L1438C56-L1438C91).

Anyway, I agree that this should be investigated in another PR, probably won't be a quick fix unfortunately.